invenio/modules/checker/tasks/tasks.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+##
+## This file is part of Invenio.
+## Copyright (C) 2013, 2014 CERN.
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Workflow tasks for checker module."""
+
+from functools import wraps, partial
+from importlib import import_module
+
+from invenio.base.utils import partial_argc
+from invenio.modules.checker.rules import Rules, Rule
+from invenio.modules.records.api import get_record, Record
+from invenio.legacy.bibsched.bibtask import task_low_level_submission
+
+from dictdiffer import diff
+import time
+import os
+import tempfile
+from invenio.config import CFG_TMPSHAREDDIR
+
+
+def _ensure_key(key, dict_):
+    if key not in dict_:
+        dict_[key] = {}
+
+
+def _store_extras(obj, extra_data, records):
+    """Update `extra_data`, with its new value and the new state of records."""
+    _ensure_key('records', extra_data)
+    for record in records:
+        extra_data['records'][record['recid']] = record.dumps()
+    obj.extra_data = extra_data
+
+
+def _load_record(extra_data, recid, local_storage_only=False):
+    """Load a record by recid.
+
+    Revives a record from `extra_data`. If the spell fails, loads it from the
+    database.
+
+    :returns: record
+    :rtype: invenio.modules.records.api.Record
+    """
+    try:
+        # Get from extra data
+        return Record(extra_data['records'][recid])
+    except KeyError as e:
+        if local_storage_only:
+            e.args += ('Non-loaded record {id_} requested. Programming error(?)'
+                       .format(id_=recid),)
+            raise e
+        # Get from the database
+        record = get_record(recid)
+        if not record:
+            raise LookupError('Record {} vanished from the databse!'
+                              .format(recid))
+        return record
+
+
+def _upload_amendments(extra_data, recids, holdingpen=False):
+    """Upload all modified records."""
+    # FIXME
+    # if task_get_option("no_upload", False) or len(records) == 0:
+    #     return
+
+    records = (_load_record(extra_data, recid, local_storage_only=True)
+               for recid in recids)
+    records_xml = ""
+    for record in records:
+        records_xml += record.legacy_export_as_marc()
+    if not records_xml:
+        return
+    records_xml = (
```

Comment:
@dset0x that's why there is a RFC #2662, where we should agree on the correct way to go (cc @tiborsimko).